napari/layers/base/base.py

```diff
@@ -802,12 +804,25 @@ def _get_base_state(self):
     def _get_state(self):
         raise NotImplementedError()
 
+    def _get_state_with_deprecated(self) -> dict:
```

Comment:
@jni , @tlambert03 : I tried a few things, but I landed on this. I realized that `Layer._get_state` is called in `_write_single_layer_with_plugins` *and* `Layer.as_layer_data_tuple`.

This change means I only need this temporary workaround in one place. All these `_get_state` methods are a little ugly, but they are private right now. My guess is that this isn't the last piece of state that we deprecate, so we might want to think about how to handle this more generally. One simple change is to make the abstract method that subclasses implement something like `_get_layer_specific_state` and make the base `Layer` just implement `_get_state` by adding the base state, appending the layer specific state, then adding any deprecated state.

Reply:
I'm starting to think the simplest thing to do is just leave `properties` and `property_choices` in the return value of `_get_state`, as well as the new `features`.

When we deprecate `properties` and `property_choices` in a follow-up PR (but continue to include them in `_get_state`), this will cause `test_layer_save.py` and `test_serialize.py` to fail due to at least one deprecation warning, but maybe ignoring or guarding against those warnings in the tests is OK?

Another option is to add a `include_deprecated` parameter to `as_layer_data_tuple` and `_get_state` that defaults to False, but is True when that state is exposed to plugins. That adds a little extra implementation to the Layers, but not much. I mostly just don't like adding a parameter that could be a temporary workaround, but maybe if Layers as EventedModels is close then this problem will go away?

@sofroniewn : could also be useful to get your perspective here or any other ideas

Reply:
> Another option is to add a include_deprecated parameter to as_layer_data_tuple and _get_state that defaults to False, but is True when that state is exposed to plugins. That adds a little extra implementation to the Layers, but not much. I mostly just don't like adding a parameter that could be a temporary workaround, but maybe if Layers as EventedModels is close then this problem will go away?

This would be the option I would choose, but I'm also ok with "just" leaving both items on the dict for now. Ultimately I don't think this is *hugely* important to get exactly right because a code search of `napari_get_reader properties` finds zero uses in the wild (only our own built-ins). So I would go with what seems easiest.

Reply:
Yeah. I think it's fine to break a little here. Once we make the layers pydantic models we'll drop `_get_state` in favour of pydantics to_dict and to_json (or similarly named) methods